File: greedy/meeting-rooms-ii.java
class Solution {
    public int minMeetingRooms(int[][] intervals) {
        int ans = 0;
        int[] arr = new int[1000001];
        for (int i = 0; i < intervals.length; i++) {
            arr[intervals[i][0]]++;
            arr[intervals[i][1]]--;
        }
        for (int j = 1; j < 1000001; j++) {
            arr[j] += arr[j-1];
            ans = Math.max(arr[j], ans);
        }
        return ans;
    }
}

File: prefix-sum/meeting-rooms-ii.py
class Solution:
    def minMeetingRooms(self, intervals: List[List[int]]) -> int:
        line = defaultdict(int)
        ans, cur = 0, 0
        for interval in intervals:
            line[interval[0]] += 1
            line[interval[1]] -= 1
        
        for k in sorted(line.keys()):
            cur += line[k]
            ans = max(cur, ans)

        return ans